share/eds/byzantine/bad_encoding.go

```diff
@@ -167,31 +185,45 @@ func (p *BadEncodingProof) Validate(hdr libhead.Header) error {
 		shares[index] = share.GetData(shr.Share)
 	}
 
-	odsWidth := uint64(len(merkleRoots) / 2)
 	codec := share.DefaultRSMT2DCodec()
 
-	// rebuild a row or col.
+	// We assume that the proof is valid in case we proved the inclusion of `Shares` but
+	// the row/col can't be reconstructed, or the building of NMTree fails.
```

Comment:
Non-blocking, can improve in a follow-up issue. Instead of "assume," IMO it's better to say we can conclude. I.e. it's the end instead of the beginning of the process.

Reply:
Fixed in [0aafbec](https://github.com/celestiaorg/celestia-node/pull/2383/commits/0aafbec72e7b7bfa2ea84d6375e936aeb1079e63)